Services/Certificate/classes/Gui/class.ilCertificateGUIFactory.php

```diff
@@ -1,20 +1,17 @@
-<?php
+<?php declare(strict_types=1);
+
 /* Copyright (c) 1998-2018 ILIAS open source, Extended GPL, see docs/LICENSE */
 
+use ILIAS\DI\Container;
+
 /**
  * @author  Niels Theen <ntheen@databay.de>
  */
 class ilCertificateGUIFactory
 {
-    /**
-     * @var
-     */
-    private $dic;
+    private ?Container $dic;
```

Comment:
This is not optional. Only the passed constructor argument is optional, the value stored in `$this->dic` MUST never be `null`.